src/pages/CommandView/form-data/get-schema/get-schema.ts

```diff
@@ -0,0 +1,337 @@
+  return {
+    ...baseResult,
+    /*
+       maximum and minimum in this context mean the greatest or smallest
+      value allowed
+     */
+    ...(parameter.maximum && parameter.type === 'Integer'
+      ? { maximum: parameter.maximum }
+      : null),
+    ...(parameter.minimum && parameter.type === 'Integer'
+      ? { minimum: parameter.minimum }
+      : null),
+    /*
+      maximum and minimum in this context mean the length of the input
+     */
+    ...(parameter.maximum && parameter.type === 'String'
+      ? { maxLength: parameter.maximum }
+      : null),
+    ...(parameter.minimum && parameter.type === 'String'
+      ? { minLength: parameter.minimum }
+      : null),
+    ...getFormat(parameter.type),
+  }
+}
+
+const extractProperties = (parameter: StrippedParameter) => {
+  const hasSubParameters = parameter.parameters.length > 0
+  const hasChoices = !(
+    typeof parameter.choices === 'undefined' || parameter.choices === null
+  )
+  const isMulti = parameter.multi
+
+  /* blow up if some assumed properties are not satisfied */
+  if (hasSubParameters && hasChoices) {
+    throw new Error('hasSubParameters AND hasChoices is not allowed')
+  }
+
+  let extractedProperties: object
+
+  if (hasSubParameters) {
+    extractedProperties = extractSubParameters(parameter)
+  } else if (hasChoices) {
+    extractedProperties = extractChoices(parameter)
+  } else if (isMulti) {
+    /* i.e. doesn't have sub params or choices */
+    extractedProperties = extractMulti(parameter)
+  } else {
+    /* doesn't have sub params, choices or multi */
+    extractedProperties = extractPlain(parameter)
+  }
+
+  return extractedProperties
+}
+
+const parametersToProperties = (parameters: Array<Parameter>) => {
+  /*
+       From an array of Parameters, create an object by hoisting up each
+       Parameter's 'key' value  to be a key in the resulting object. then set
+       the value of that key to be a transformation of the remaining values
+       of the Parameter.
+    */
+  return parameters.reduce((accumulator, parameter) => {
+    const { key, ...rest } = parameter
+    return { ...accumulator, [key]: { ...extractProperties(rest) } }
+  }, {})
+}
+
+const getRequired = (parameters: Parameter[]) => {
+  return parameters.filter((x) => !x.optional).map((x) => x.key)
+}
+
+const getParameterSchema = (parameters: Array<Parameter>) => {
+  /*
+    Derive parameter schema -- factored into a function so that it can be
+    reused by subparameters.
+   */
+  return {
+    type: 'object',
+    properties: parametersToProperties(parameters),
+    required: getRequired(parameters),
+  }
+}
+
+const getSchema = (
+  instances: Array<Instance>,
+  parameters: Array<Parameter>,
+) => {
+  type instanceList = (string | undefined)[]
+  type enumType = { enum: instanceList } | { default: string }
+
+  let instancesData: enumType | null = null
+
+  if (instances.length > 1) {
+    const enumArray: instanceList = [''].concat(instances.map((x) => x.name))
```

Comment:
That was on purpose because the library docs told me it was necessary to have a dropdown without a default have an empty string as it's first entry. They lied lol. Just refactored to remove this entirely and it appears to work just fine.